File: Domain/MoneyMaster.Domain.Entities/Entities/Account.cs
namespace MoneyMaster.Domain.Entities
{
    /// <summary>Счет пользователя</summary>
    /// <typeparam name="TKey">Тип первичного ключа</typeparam>
    public class Account<TKey> : NamedTimedEntity<TKey>
    {
        /// <summary>Баланс</summary>
        public decimal Balance { get; set; }

        /// <summary>Расходы</summary>
        public string? Currency { get; set; }

        /// <summary>Иконка</summary>
        public string? Icon { get; set; }

        /// <summary>Идентификатор пользователя</summary>
        public TKey? UserId { get; set; }

        /// <summary><inheritdoc cref="MoneyMaster.Domain.Entities.User"/></summary>
        public User? User { get; set; }

        /// <summary>Мягкое удаление</summary>
        public bool IsDelete { get; set; }

        /// <summary>Идентификатор типа счета записи</summary>
        public TKey? AccountTypeId { get; set; }

        /// <summary><inheritdoc cref="MoneyMaster.Domain.Entities.AccountType"/></summary>
        public AccountType? AccountType { get; set; }

        /// <summary>Коллекция отчетов </summary>
        public ICollection<Report>? Reports { get; set; }

        /// <summary>Коллекция транзакций </summary>
        public ICollection<Transaction>? Transactions { get; set; }
    }

    /// <summary><inheritdoc/></summary>
    public class Account : Account<Guid> { }
}

File: MoneyMasterFrontend/frontend/src/types/account.ts
export interface AccountResponse {
  id: string;
  name: string;
  balance: number;
  currency: string;
  icon: string;
  userId: string;
  accountTypeId: string;
  createAt: string;
}

export interface AccountType {
  id: string;
  name: string;
  icon: string;
  isSystem: boolean;
  isDelete: boolean;
  createAt: string;
}

export interface AccountCreateInfo {
  accountTypes: AccountType[];
  currentCode: string[];
}

export interface UpdatingAccountRequest {
  id: string;
  name: string;
  balance: number;
  currency: string;
  icon: string;
  userId: string;
  accountTypeId: string;
}
